src/fillTank.js

```diff
@@ -14,17 +14,38 @@
  * @param {number} amount
  */
 function fillTank(customer, fuelPrice, amount = Infinity) {
-  const { vehicle } = customer;
+  const { vehicle, money } = customer;
+
+  if (typeof customer !== 'object'
+    || customer === null || Array.isArray(customer)) {
```

Comment:
The condition `typeof customer !== 'object' || customer === null || Array.isArray(customer)` is correct for checking if `customer` is a valid object. However, it might be more readable to use `!(customer && typeof customer === 'object' && !Array.isArray(customer))`.

Reply:
done